app/crud/base.py

```diff
@@ -131,6 +132,65 @@ def get_all(
         query = self.order_by(db.query(self.model), order_by=order_by)
         return query.all()
 
+    def search(
+        self,
+        db: Session,
+        *,
+        search_column: str,
+        search_term: str,
+        order_by: Optional[List[str]] = None,
+        limit: int = 0,
+    ) -> List[ModelType]:
+        """search all the records from the database based on the search column,
+        search term then order the results by particular column and then return
+        the first limited value results.
+
+        Parameters
+        ----------
+        db : Session
+            The database session.
+        search_column : str
+            This string specifies which columns to search the search_term for.
+        search_term : str
+            The string to use to search for in the search_column either startswith
+            or endswith or a fuzzy search.
+        order_by : Optional[List[str]], optional
+            List of column names to order by. If a column name is prefixed with '-',
+            order it in descending order.
+        limit : int, optional
+            This value controls the number of results returned, by default 0.
+
+        Returns
+        -------
+        List[ModelType]
+            A list of SQLAlchemy model instances from the query.
+        """
+        if hasattr(self.model, search_column):
+            # Keeping this as a comment so as to get back to the previous exact match
+            # system for stable response purposes.
+
+            # query = db.query(self.model).filter(
+            #     getattr(self.model, search_column).contains(search_term)
+            # )
+            # SIMILARITY function will work only if the extension is enabled.
+            # only works with string data
+            similarity_func = func.similarity(
+                getattr(self.model, search_column), search_term
+            )
+            query = (db.query(self.model).where(similarity_func > 0.1)).order_by(
```

Comment:
I updated this part so we only get the results that are at least 0.1 similar to the search term.
I also changed the `else` part to throw 400 when searching against a non-existing column.